### PR TITLE
fix: use alternate scheme for importing multipart

### PR DESCRIPTION
**Pull request by henryiii**
This is an alternate attempt at #166 that allows a user to install the package and import it in a single Python process (not recommended, but Google Colab basically requires this to install anything).

This has the following pro's:

* Doesn't rely on the import system loading a `.pth` file, meaning it can be hot swapped
* Editable installs are supported (ironically, editable installs use the same `.pth` and custom loader that the original attempt did, but Google Colab users don't do editable installs!)
* Less invasive on `import multipart` if multipart is not installed.

And the following con's:

* Only handles filesystems; if the imports are not from file systems, it will not work (it will have the old collision behavior in that case).
    * This could be fixed if it's a problem, I think, but I think this is fine unless you have users loading both packages from a non-filesystem (like a zip).
* A bit more invasive on `multipart`, specifically requiring that package not be moved to a module structure while this workaround is in place.

I also added a wrapper script for nox like the other scripts in `/scripts`.

### CODE REVIEW

**Comment by Kludex** [comment]
@defnull you good with the last point on the cons?

**Comment by defnull** [comment]
I have no plans to turn `multipart.py` into a package anytime soon. Since this is not a permanent thing but just a temporary workaround, I'm absolutely fine with that :)

**Comment by defnull** [comment]
A quick test looks promising:

```sh
$ python3 -mvenv venv
$ venv/bin/pip install git+https://github.com/henryiii/python-multipart.git@henryiii/fix/importer
[...] Successfully installed python-multipart-0.0.13

$ venv/bin/python -c "import multipart; print(multipart.__file__)"
<string>:1: FutureWarning: Please use `import python_multipart` instead.
/home/marc/git/multipart/conflict-tests/venv/lib/python3.12/site-packages/multipart/__init__.py

$ venv/bin/python -c "import python_multipart as multipart; print(multipart.__file__)"
/home/marc/git/multipart/conflict-tests/venv/lib/python3.12/site-packages/python_multipart/__init__.py

$ venv/bin/pip install multipart
[...] Successfully installed multipart-1.1.0

$ venv/bin/python -c "import python_multipart as multipart; print(multipart.__file__)"
/home/marc/git/multipart/conflict-tests/venv/lib/python3.12/site-packages/python_multipart/__init__.py

$ venv/bin/python -c "import multipart; print(multipart.__file__)"
/home/marc/git/multipart/conflict-tests/venv/lib/python3.12/site-packages/multipart.py
```

**Comment by defnull** [comment]
This also works as expected in Google Collab:
image

The last edge case "conflict while importing from zip-files or other non-file-system imports" should be exceptionally rare and the important parts of this PR (being backwards compatible and printing a warning) should still work, so it's still an improvement over the current state.

**Comment by merwok** [comment]
Is this rework needed? The original problem came from installing a new module and trying to import it in the same process _without_ invalidating importlib caches

**Comment by henryiii** [comment]
Colab users don't know they need to invalidate importlib caches, and the help in colab doesn't tell them they need to. And invalidating the import caches doesn't work, since the problem is the `.pth` file is run at startup, and is not related to import caches. Remember, the problem can be seen with:

```python
!pip install python-multipart==0.0.13
import multipart
```

The `.pth` files are run when the notebook kernel starts, which doesn't have any dependencies at all (Colab doesn't give you a way to pre-install deps). Then you install the package (which adds the .pth file), but it never gets a chance to run. Adding an `importlib.invalidate_caches()` in-between has no effect.

FWIW, this works, because it does exactly what the `.pth` file would have done:

```python
!pip install python-multipart==0.0.13
import _python_multipart_loader
import multipart
```

But no one would know to do that.

**Comment by henryiii** [comment]
If you restart the session, Colab does keep the environment so it works the second time. Also, this works:

```python
!pip install python-multipart==0.0.13
import site
site.main()
import multipart
```

But again, no one would know they need to do that.

File: scripts/README.md [hunk 1]
```diff
@@ -4,5 +4,6 @@
 * `scripts/test` - Run the test suite.
 * `scripts/lint` - Run the code format.
 * `scripts/check` - Run the lint in check mode, and the type checker.
+* `scripts/rename` - Check that the backward-compat `multipart` name works as expected
```
Comment (Kludex) [inline scripts/README.md:7]:
```suggestion
* `scripts/rename` - Check that the backward-compat `multipart` name works as expected.
```

**Review by Kludex — APPROVED** [review]

**Comment by Kludex** [comment]
Thanks. Let's try this.

**Comment by Kludex** [comment]
Yeah... Me neither... 🤔

**Comment by defnull** [comment]
The build artefacts uploaded to pypi do not contain the second package for some reason. Installing directly from git also does not work, but installing from `git+https://github.com/henryiii/python-multipart.git@henryiii/fix/importer` does in fact still work. What!?

**Comment by Kludex** [comment]
I'm yanking python-multipart 0.0.14 and Starlette 0.41.1.

**Comment by henryiii** [comment]
Ah, forgot to rerun `check-sdist`. Sorry, will fix!
